face_api_dataset/dataset.py

```diff
@@ -955,3 +1001,47 @@ def _read_landmarks_3d(cls, info: dict, mdt: Modality) -> Dict[Id, Landmark_3D]:
             landmarks[lmk_name] = tuple(landmark["camera_space_pos"])
 
         return landmarks
+
+    def get_with_camera_and_frame(self, i: int, camera_name: str = "default", frame_no: int = 1) -> dict:
+        """Returns the modalities for a given i, camera_name, frame_no."""
+        if i > len(self):
+            raise ValueError(f"Index {i} is out of bounds")
+        number = self._image_numbers[i]
+        info = None
+        if self._needs_info:
+            info_file = self._root / \
+                f"{_Extension.filename_prefix_for_camera_and_frame(number, camera_name, frame_no)}.{_Extension.INFO}"
+            with info_file.open("r") as f:
+                info = json.load(f)
+
+        ret = {}
+        for modality in self._modalities:
+            ret[modality] = self._open_modality(
+                modality, number, info, camera_name, frame_no)
+        return ret
+
+    def render_number_at(self, i: int) -> str:
+        """Returns the internal value of the render number at a dataset index"""
+        if i > len(self):
+            raise ValueError(f"Index {i} is out of bounds")
+        return self._image_numbers[i]
+
+    def info_files_for_render_id(self, number: str) -> Generator[Path, None, None]:
+        """Returns a glob match for info files with this render number (all cameras and frames)"""
+        return self._root.glob(f"{number}.*{_Extension.INFO}")
+
+    def info_files_for_render_id_with_camera(self, number: str, camera_name: str) -> Generator[Path, None, None]:
+        """Returns a glob match for info files with this camera name and render number"""
+        return self._root.glob(f"{_Extension.filename_prefix_for_camera(number, camera_name)}.*{_Extension.INFO}")
+
+    def unique_camera_names_for_render_id(self, number: str) -> List[str]:
+        """Returns an array of unique camera names for this render id, like ['my_cam_one', 'my_cam_another', 'my_cam_example']"""
+        return [_Extension.camera_name_from_file_name(f.name) for f in set(self.info_files_for_render_id(number))]
+
+    def camera_frame_nos_map_for_render_id(self, number: str) -> Dict[str, List[int]]:
```

Comment:
I would rather add camera map as a property, to maintain the same style as segments. Also I think camera should be added as yet another modality.